src/Symfony/Bundle/FrameworkBundle/EventListener/SuggestMissingPackageSubscriber.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\FrameworkBundle\EventListener;
+
+use Symfony\Component\Console\ConsoleEvents;
+use Symfony\Component\Console\Event\ConsoleErrorEvent;
+use Symfony\Component\Console\Exception\NamespaceNotFoundException;
+use Symfony\Component\EventDispatcher\EventSubscriberInterface;
+
+/**
+ * Suggests a package, that should be installed (via composer),
+ * if the package is missing, and the input command namespace can be mapped to a Symfony bundle.
+ *
+ * @author Przemysław Bogusz <przemyslaw.bogusz@tubotax.pl>
+ */
+final class SuggestMissingPackageSubscriber implements EventSubscriberInterface
+{
+    /**
+     * Mappings between namespaces of Symfony commands and packages required to run these commands.
+     *
+     * @var array
+     */
+    private static $suggestedPackages = [
+        'doctrine' => [
+            'Doctrine ORM',
+            'symfony/orm-pack',
+        ],
+        'generate' => [
+            'SensioGeneratorBundle',
+            'sensio/generator-bundle',
+        ],
+        'make' => [
+            'MakerBundle',
+            'symfony/maker-bundle --dev',
+        ],
+        'server' => [
+            'Symfony Web Server',
+            'symfony/web-server-bundle --dev',
+        ],
+    ];
+
+    public function onConsoleError(ConsoleErrorEvent $event)
+    {
+        if ($event->getError() instanceof NamespaceNotFoundException) {
+            if (null !== $event->getInput()->getFirstArgument()) {
+                $namespace = explode(':', $event->getInput()->getFirstArgument())[0];
+
+                if (isset(self::$suggestedPackages[$namespace])) {
+                    $suggestion = self::$suggestedPackages[$namespace];
+                    $message = sprintf("It seems that you are trying to run a command from '%s' bundle/package, but you do not have it installed. You may try installing the missing bundle/package by running:\n\n    composer require %s", $suggestion[0], $suggestion[1]);
```

Comment:
Exceptions message should never be formatted, `Application::renderException()` already takes care of the output. 
Here is a proposal that is consistent with existing DX error messages:
```php
$message = sprintf('%s You may be looking for a command provided by the %s which is not installed currently. Try running "composer require %s".', $e->getMessage(), $suggestion[0], $suggestion[1]);
```

Which gives:
> There are no commands defined in the "make" namespace. You may be looking for a command provided by the MakerBundle which is not installed currently, try running "composer require symfony/maker-bundle --dev".

Reply:
By formatted you mean adding new lines?

Reply:
Nevermind